contracts/DealProvider/IDealProvierEvents.sol

```diff
@@ -10,7 +10,7 @@ interface IDealProvierEvents {
         address token;
     }
 
-    event NewPoolCreated(BasePoolInfo poolInfo, uint256[] params);
+    event NewPoolCreated(uint256 indexed poolId, address indexed owner, address indexed token, uint256[] params);
```

Comment:
The **indexed** keyword in Solidity events is primarily used for parameters that are frequently used for filtering or querying purposes. In the case of the `poolId` parameter in your `NewPoolCreated` event, it's generally not necessary or beneficial to add the **indexed** keyword. Indexing the `poolId` would result in additional storage costs and may not provide significant advantages in terms of event filtering or searching. 
lets change it to:
```solidity
event NewPoolCreated(uint256 poolId, address indexed owner, address indexed token, uint256[] params);
```

Reply:
> @YouStillAlive Indexing the `poolId` doesn't make any disadvantages to us.
> 
> 1. We can filter the data by the `poolId` later. It doesn't affect the filtering.
> 2. Indexing the 'poolId` reduces the gas.
> 
> So I indexed the `poolId. What do you think?

@0xSolcroot While I understand your perspective, I tend to agree with @YouStillAlive. Given that the range of the poolId is from 0 to 2^256, the inherent indexing of this data type might be sufficient. Adding an explicit index to poolId could lead to unnecessary gas costs. This is because every indexed property needs its own separate space in the logs. Therefore, unless we have a specific reason to improve the search efficiency for poolId through indexing, it might be more cost-effective to leave it as is.

Reply:
@0xSolcroot I cannot imagine a scenario where we will need to filter the `NewPoolCreated` event with `poolId`. However, I'm curious to know how indexing `poolId` will reduce gas?

Reply:
@Lomet @ashwinarora Indexing the 'poolId' reduce the gas.
https://0xmacro.com/blog/solidity-gas-optimizations-cheat-sheet/

Reply:
Of course, we might need to filter the event with the `poolId` but indexing that field would be helpful in terms of the gas optimization.

Reply:
yes, let's add an indexed keyword